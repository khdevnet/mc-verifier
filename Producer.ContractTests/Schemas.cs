using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using McVerifier;
using Xunit;

namespace Producer.ContractTests;

[UseApprovalSubdirectory("Approvals")]
[UseReporter(typeof(VisualStudioReporter))]
public class Schemas
{
    [Fact]
    public void UserCreatedEvent()
        => MessageSchema
            .Create(typeof(UserCreatedEvent))
            .Verify();

    [Fact]
    public void CreateUserCommand()
        => MessageSchema
            .Create(typeof(CreateUserCommand))
            .Verify();
}
