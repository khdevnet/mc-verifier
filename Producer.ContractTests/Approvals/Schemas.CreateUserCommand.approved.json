{
    
  "$schema":  "http://json-schema.org/draft-04/schema#",
    
  "title":  "CreateUserCommand",
    
  "type":  "object",
    
  "required":  [
        
    "Id",
        
    "Name",
        
    "Price"
  
    ],
    
  "properties":  {
        
    "Id":  {
            
      "type":  "integer",
            
      "format":  "int32",
            
      "maximum":  10.0,
            
      "minimum":  0.0
    
        },
        
    "Name":  {
            
      "type":  "string",
            
      "minLength":  1
    
        },
        
    "Price":  {
            
      "type":  "number",
            
      "format":  "decimal",
            
      "maximum":  10.0,
            
      "minimum":  0.0
    
        }
  
    }

}